controllers/disruption_controller.go

```diff
@@ -275,7 +274,7 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 
 		// check if we have reached trigger.createPods. If not, skip the rest of reconciliation.
-		requeueAfter := time.Until(helpers.TimeToCreatePods(instance.Spec.Triggers, instance.CreationTimestamp.Time))
+		requeueAfter := time.Until(instance.TimeToCreatePods(instance.CreationTimestamp.Time))
```

Comment:
Are using anything else than `instance.CreationTimestamp`?

If not, we could remove this argument